test_infra/src/lib.rs

```diff
@@ -798,6 +798,22 @@ pub fn check_matched_lines_count(input: &str, keywords: Vec<&str>, line_count: u
     }
 }
 
+pub fn kill_child(child: &mut Child) {
+    let r = unsafe { libc::kill(child.id() as i32, libc::SIGTERM) };
+    if r != 0 {
+        let e = io::Error::last_os_error();
+        if e.raw_os_error().unwrap() == libc::ESRCH {
+            return;
+        }
+        eprintln!("Failed to kill child with SIGTERM: {e:?}");
+    }
+
+    // The timeout period elapsed without the child exiting
+    if child.wait_timeout(Duration::new(5, 0)).unwrap().is_none() {
```

Comment:
I'm a bit worried about this as we also call `.wait_without_output` later - will this earlier call "lose" the output?

Reply:
I test with the following case, the messages could output whether child times out.

```rust
fn main() {
    let mut i = 0;
    while i < 6 {
        println!("Hello, world!");
        sleep(std::time::Duration::from_secs(1));
        i += 1;
    }
}

#[cfg(test)]
mod tests {
    use std::process::{Command, Stdio};
    use wait_timeout::ChildExt;
    #[test]
    fn it_works() {
        let mut child = Command::new("target/debug/demo")
            .stderr(Stdio::piped())
            .stdout(Stdio::piped())
            .spawn()
            .unwrap();
        if child
            .wait_timeout(std::time::Duration::from_secs(5))
            .unwrap()
            .is_none()
        {
            let _ = child.kill();
            eprintln!("Child process timed out");
        }
        let output = child.wait_with_output().unwrap();
        println!(
            "\n\n==== Start child stdout ====\n\n{}\n\n==== End child stdout ====",
            String::from_utf8_lossy(&output.stdout)
        );
        println!(
            "\n\n==== Start child stderr ====\n\n{}\n\n==== End child stderr ====",
            String::from_utf8_lossy(&output.stderr)
        );
    }
}

```